pkg/server/message.go

```diff
@@ -258,26 +254,30 @@ func (t *TranslateServiceServer) UpdateMessages(
 		}
 
 		// Update messages for all translations
-		for i := range all {
-			if all[i].Original {
-				continue
-			}
-
+		for i := range others {
 			err = t.repo.SaveMessages(ctx, params.serviceID, &all[i])
-			switch {
-			default:
-				// noop
-			case errors.Is(err, repo.ErrNotFound):
-				return nil, status.Errorf(codes.NotFound, "service not found")
-			case err != nil:
-				return nil, status.Errorf(codes.Internal, "")
+			messages, err, done = t.handleErrors(err)
+			if done {
+				return messages, err
 			}
 		}
 	}
 
 	return messagesToProto(params.messages), nil
 }
 
+func (t *TranslateServiceServer) handleErrors(err error) (*translatev1.Messages, error, bool) {
+	switch {
+	default:
+		// noop
+	case errors.Is(err, repo.ErrNotFound):
+		return nil, status.Errorf(codes.NotFound, "service not found"), true
+	case err != nil:
+		return nil, status.Errorf(codes.Internal, ""), true
+	}
+	return nil, nil, false
+}
```

Comment:
almost the same as `SaveMsgsAndHandleErrors` ?
There must be a way to combine them